transfer_nlp/plugins/predictors.py

```diff
@@ -0,0 +1,113 @@
+import inspect
+import logging
+from itertools import zip_longest
+from typing import Dict, List, Any
+
+import torch
+from ignite.utils import convert_tensor
+
+from transfer_nlp.loaders.vectorizers import VectorizerNew
+from transfer_nlp.plugins.config import register_plugin
+from transfer_nlp.plugins.helpers import ObjectHyperParams
+from transfer_nlp.plugins.trainers import BasicTrainer
+
+name = 'transfer_nlp.plugins.predictor'
+logging.getLogger(name).setLevel(level=logging.INFO)
+logger = logging.getLogger(name)
+
+
+def _prepare_batch(batch: Dict, device=None, non_blocking: bool = False):
+    """Prepare batch for training: pass to a device with options.
+
+    """
+    result = {key: convert_tensor(value, device=device, non_blocking=non_blocking) for key, value in batch.items()}
+    return result
+
+
+@register_plugin
+class PredictorHyperParams(ObjectHyperParams):
+
+    def __init__(self, trainer: BasicTrainer):
+        super().__init__()
+        self.vectorizer = trainer.dataset_splits.vectorizer
+        self.model = trainer.model
+
+
+class Predictor:
```

Comment:
should this be an ABC class? I would expect a `predict` method from a class called `Predictor`. Also given that usually pytorch forward methods leave off the final activation function to optimize training, maybe the predictor could take in a function to apply to the output? or that's too much complexity to be worth the small gain?

Reply:
Yes, this should be an ABC class,
for now the `infer` method does just what the `infer` method from the nn.Module does, and the `decode` method is an abstract method that one has to implement depending on the use case